src/UrqlClient.re

```diff
@@ -486,3 +486,51 @@ let createRequestOperation = (~client, ~operationType, ~request, ~opts=?, ()) =>
 external dispatchOperation:
   (~client: t, ~operation: ClientTypes.operation) => unit =
   "dispatchOperation";
+
+[@bs.send]
+external queryJs:
+  (
+    ~client: t,
+    ~query: string,
+    ~variables: Js.Json.t=?,
+    ~context: ClientTypes.partialOperationContextJs=?,
+    unit
+  ) =>
+  Wonka.Types.sourceT('response) =
+  "query";
+
+let query =
+    (
+      ~client: t,
+      ~request: UrqlTypes.request('response),
+      ~opts: option(ClientTypes.partialOperationContext)=?,
+      (),
+    ) => {
+  executeQuery(~client, ~request, ~opts?, ())
+  |> Wonka.take(1)
+  |> Wonka.toPromise;
```

Comment:
Does it make more sense maybe not to convert these to promises? It seems to me like you wouldn’t gain much in Reason from using promises 😅

Reply:
Agreed that I don't think we necessarily gain much, but I think it's still good to cover in the lib in case people want to interact with the `Js.Promise.t` type rather than `Wonka.sourceT`. I could imagine folks coming from JS and learning about `Js.Promise.t` and wanting a way to get a `Js.Promise.t` from these methods. These bindings are such low cost anyway that I think it's still worthwhile (tho I prefer `Wonka.subscribe` personally as well).